src/Cake.Core/CakeTask.cs

```diff
@@ -35,165 +30,114 @@ public abstract class CakeTask : ICakeTaskInfo
         /// Gets the task's dependencies.
         /// </summary>
         /// <value>The task's dependencies.</value>
-        public IReadOnlyList<CakeTaskDependency> Dependencies => _dependencies;
+        public List<CakeTaskDependency> Dependencies { get; }
```

Comment:
`IList`... Or is `ISet` or `ICollection` sufficient?

Reply:
We could probably change this to an `ICollection`.

Reply:
Changed my mind here, would probably be best to keep compatibility here and use an `IList`.

Reply:
Actually, let's keep this a `List`. No reason to expose any other type here since it can't be replaced.